### PR TITLE
tools: add `ziglint` and `zig-doctest`

### PR DESCRIPTION
**Pull request by chenrui333**
- add `ziglint` and `zig-doctest`
- refresh toc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ This repository lists some awesome projects written in Zig, maintained by [ZigCC
 <!-- toc -->
 
 - [Summary](#summary)
-  - [Community](#community)
-  - [Learning Resources](#learning-resources)
+  * [Community](#community)
```
Comment (jiacai2050) [inline README.md:10]: Those toc is generated with `make toc`.

**Comment by jiacai2050** [comment]
@chenrui333 CI failed 😢

File: README.md (hunk 1 above)
Comment (chenrui333) [reply README.md]: make sense :)

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**